### PR TITLE
Add support for Marvell MVEBU based devices (Armada,Globalscale,Linksys)

### PR DESCRIPTION
**Pull request by torte71**
https://wiki.openwrt.org/toh/linksys/wrt_ac_series
Adds new TARGET=mvebu-generic with PROFILES for
  armada-370-db
  armada-370-rd
  armada-385-db-ap
  armada-388-clearfog-base
  armada-388-clearfog-pro
  armada-388-rd
  armada-xp-db
  armada-xp-gp
  globalscale-mirabox
  linksys-wrt1200ac
  linksys-wrt1900ac
  linksys-wrt1900acs
  linksys-wrt1900acv2
  linksys-wrt3200acm

### CODE REVIEW

**Comment by SvenRoederer** [comment]
can you update the link of the OpenWRT-Wiki to the current location?

**Comment by torte71** [comment]
- Link to openwrt wiki is now corrected
- None of the modules that you mentioned are required for building the images,
except these two are required for linksys-wrt3200acm:
CONFIG_PACKAGE_kmod-mwifiex-sdio=m
CONFIG_PACKAGE_kmod-btmrvl=m
I only included them for consistency, as we have them in (almost) any of our *.config

**Comment by SvenRoederer** [comment]
in this TARGET-specific configs should only be the options required for this special TARGET (https://github.com/freifunk-berlin/firmware/blob/master/Makefile#L120).

when there should be general support for these wifi-chips, esp. the ralink-usb, these options should be moved into the common.config. 
I would prefer to have them removed from this and possibly all other TARGETS. in case of ptoblems, we can add them later but start with a minimal config for this new TARGET.

**Comment by SvenRoederer** [comment]
I merged locally and it fails to build
>  make[3] package/install
 make[3] target/install
 make[4] -C target/linux install
 make[4] -C target/sdk install
 make[4] -C target/imagebuilder install
 make[3] package/index
 make[3] checksum
make[1]: Verzeichnis „/mnt/hosts/build/src/openwrt/freifunk/lede/openwrt“ wird verlassen
touch .stamp-compiled
IB_FILE not defined; assuming called from inside regular build
ls: Zugriff auf '/mnt/hosts/build/src/openwrt/freifunk/lede/openwrt/bin/targets/mvebu/generic/*-imagebuilder-*.tar.xz' nicht möglich: Datei oder Verzeichnis nicht gefunden
mkdir -p /mnt/hosts/build/src/openwrt/freifunk/lede/firmwares/mvebu-generic
./assemble_firmware.sh -p "armada-370-db armada-370-rd armada-385-db-ap armada-388-clearfog-base armada-388-clearfog-pro armada-388-rd armada-xp-db armada-xp-gp globalscale-mirabox linksys-wrt1200ac linksys-wrt1900ac linksys-wrt1900acs linksys-wrt1900acv2 linksys-wrt3200acm" -i  -e /mnt/hosts/build/src/openwrt/freifunk/lede/embedded-files -t /mnt/hosts/build/src/openwrt/freifunk/lede/firmwares/mvebu-generic -u "default tunnel-berlin backbone"
IB_FILE does not exist -e
Makefile:207: die Regel für Ziel „.stamp-images“ scheiterte
make: *** [.stamp-images] Fehler 1

File: profiles/mvebu-generic.profiles [hunk 1]
```diff
@@ -0,0 +1,14 @@
+armada-370-db
+armada-370-rd
+armada-385-db-ap
+armada-388-clearfog-base
+armada-388-clearfog-pro
```
Comment (SvenRoederer) [inline profiles/mvebu-generic.profiles]: From my former researches I remember the ClearFrog boards are only PCBs w/o chassis or PSU. 
Beside the Linksys boards all others seem to be some kind of Developmentboards and no "enduser products", so not sure if it makes sense to build these

**Comment by torte71** [comment]
I overlooked, that the "armada-388-clearfog-base" and "armada-388-clearfog-pro" were split more recently, in "master" there was only "armada-388-clearfog".
As these are all testing boards only, I removed them as suggested.

**Comment by booo** [comment]
@torte71 can you give us an update on the state of the branch? If it works we can merge it into master.

**Comment by SvenRoederer** [comment]
I found some old commits in my repo, which fix the cortex-support (c17bb733011fe3d6e648d6187d7e935ee9a67f03).
I also had some code to add the Turris Omnia board.

**Pull request closed without merging**